File: Shared/Models/Schedule/Schedule.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Shared.Models.ScheduleGroups;

namespace Shared.Models.Schedule;

public class Schedule
{
	[Key] public int Id { get; set; }

	[ForeignKey(nameof(Course))] public int CourseId { get; set; }

	[ForeignKey(nameof(Professor))] public int ProfessorId { get; set; }


	/// Room № ? or remote 
	public string? Location { get; set; }
	
	// public ICollection<Group> Groups { get; set; } = new List<Group>();
	
	public ICollection<ScheduleGroup> ScheduleGroups { get; set; } = new List<ScheduleGroup>();
	
	public Course Course { get; set; } = null!;

	public Professor Professor { get; set; } = null!;


	// Represent in Unix TimeStamp
	public long StartTime { get; set; }
	public long EndTime { get; set; }


}

File: lib/model/schedule/schedule.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import '../course/course.dart';
import '../group/group.dart';
import '../professor/professor.dart';

part 'schedule.freezed.dart';

part 'schedule.g.dart';


@freezed
class Schedule with _$Schedule {
  const factory Schedule({
    required int id ,
    required String location,
    required Course course,
    required List<Group> groups,
    required Professor professor,
    required DateTime startTime,
    required DateTime endTime,
  }) = _Schedule;

  factory Schedule.fromJson(
    Map<String, dynamic> json,
  ) =>
      _$ScheduleFromJson(json);
}
